### PR TITLE
HHH-14346 Check declaredPluralAttributes for null before access

### PR DESCRIPTION
**Pull request by MikeEdgar**
Fixes HHH-14346

### CODE REVIEW

**Comment by MikeEdgar** [comment]
Can one of the project team members please rerun the build? I believe the failure is unrelated to the changes. Please also let me know if this needs to be altered in some way.

File: hibernate-core/src/test/java/org/hibernate/jpa/test/metadata/MetadataTest.java [hunk 1]
```diff
@@ -268,6 +268,13 @@ public void testCollection() throws Exception {
 		assertTrue( flowers instanceof ListAttribute );
 	}
 
+	@Test
```
Comment (beikov) [inline hibernate-core/src/test/java/org/hibernate/jpa/test/metadata/MetadataTest.java:271]: Please add `@TestForIssue(jiraKey = "HHH-14346")`

**Comment by beikov** [comment]
Please create a PR against the master branch. I'll make sure that we consider to backport this fix to 5.4

**Comment by Sanne** [comment]
Many thanks, I've applied it already and included @beikov 's suggestion, as we'll release a 5.4 soon :)

**Pull request closed without merging**